src/zmq/zmqpublishnotifier.cpp

```diff
@@ -248,9 +253,12 @@ bool CZMQPublishRawBlockNotifier::NotifyBlock(const CBlockIndex *pindex)
     LogPrint(BCLog::ZMQ, "Publish rawblock %s to %s\n", pindex->GetBlockHash().GetHex(), this->address);
 
     const Consensus::Params& consensusParams = Params().GetConsensus();
+    BlockManagerOpts blockman_opts{};
+    Assert(!ApplyArgsManOptions(gArgs, blockman_opts)); // already checked on init
+    BlockManager blockman{blockman_opts};
```

Comment:
I may have missed some nuance here. To me it seemed acceptable, because functions are protected by `cs_main`. For const functions like`ReadBlockFromDisk` this seems safe to me (though I'm still new to the locking model and might be missing something here). However doing block write and pruning operations with two blockmanagers is fragile, since their internal state might no longer reflect the state on disk. 

I'm not sure how to proceed here. I could mark the `BlockManager` here as const, that would at least ensure that it's state cannot change.

Reply:
> I may have missed some nuance here. To me it seemed acceptable, because functions are protected by cs_main. For const functions likeReadBlockFromDisk this seems safe to me

Note that according to the PR description in #27006 there is currently at least one `cs_main` related bug in `ReadBlockFromDisk`.

Reply:
> I feel this though, I'll try out the first and second option you laid out.

Both are branching out a bit too much in my taste for the refactor we seek to achieve here. If the `const BlockManager` is a non-starter, I can always revert back to https://github.com/bitcoin/bitcoin/commit/f87c39399823e22c553b797cc66fa4063462a32b

Reply:
This patch is much simpler than what I came up with, thanks a lot!